src/main/java/it/gov/pagopa/payhub/auth/utils/JWTValidator.java

```diff
@@ -86,4 +90,27 @@ public void validateInternalToken(String token) {
             throw new InvalidTokenException("The token is not valid");
         }
     }
+
+    /**
+     * Validates JWT signature with publickey.
+     *
+     * @param applicationName the application name for which to validate the token
+     * @param token the JWT to validate
+     * @param publicKey the key to use in the verify instance.
+     * @throws Exception if the token is invalid for any other reason
+     *         (e.g., signature verification failure).
+     */
+    public Pair<String, Map<String, Claim>> validateLegacyToken(String applicationName, String token, PublicKey publicKey) {
```

Comment:
for a Utility class, to have a specific method like this is something with no value outside the legacy token use cases...

instead of a `validateLegacyToken`, let's just implement here an overloaded version of the `validate` method, which will accept the token and the publicKey.

The `validateLegacyToken` with the Pair construction or null will be implemented inside the JwtLegacyValidatorService (the only place where it's supposed to be used this logic)